src/egl_context.c

```diff
@@ -175,6 +175,14 @@ static GLFWbool chooseEGLConfig(const _GLFWctxconfig* ctxconfig,
         u->samples = getEGLConfigAttrib(n, EGL_SAMPLES);
         u->doublebuffer = GLFW_TRUE;
 
+#if defined(_GLFW_WAYLAND)
+        // Avoid using transparent buffer on Wayland if transparency is not requested.
+        // Otherwise mutter will fail to properly screenshot OpenGL content.
```

Comment:
Oh, it does screenshot properly, it's just does what you ask of it. If you ask for a surface with an alpha channel, we'll create a screenshot using the same alpha channel intact. The opaque region is irrelevant, the only thing it affects is how a blitting shader is set up when compositing.

Reply:
@jadahl I have no idea what that means but that's clearly counterintuitive because it displays or screenrecords just fine and only screenshots are broken. is there alternative fix you wanted to suggest?

Reply:
If you record the screen, you'll record the composited content. If you record the window itself, you should get it with the alpha channel intact, but AFAIK not many video codecs support that so it'll treat is as opaque/black. Screenshots turn into PNGs though, which supports the alpha channel, so it'll stay intact.

I think the change is correct, not only for Wayland but for any EGL context, only the comment is incorrect.